src/Compiler/SyntaxTree/SyntaxTrivia.fsi

```diff
@@ -384,3 +384,11 @@ type SynArgPatsNamePatPairsTrivia =
         /// The syntax range from the beginning of the `(` token till the end of the `)` token.
         ParenRange: range
     }
+
+/// Represents additional information for SynType.Or
+[<NoEquality; NoComparison>]
+type SynTypeOrTrivia =
+    {
+        /// The syntax range of the `or` keyword
+        OrKeyword: range
```

Comment:
Is this sufficient?
What if the chain contains multiple OR statements, is that covered?

Reply:
(I assume further ORs will be nested, always treating each as a binary operator.. ? )

Reply:
Yes, the ORs will be nested on the left-hand side.
That is why I added `TypesForTypar` to collect the list of types.